### ISSUE TITLE
icicle-test: x64.ins test failure `ST0`

### ISSUE THREAD

**Issue by ekilmer** [icicle-emu/icicle-emu#54]
On latest commit ([e0da76a](https://github.com/icicle-emu/icicle-emu/commit/e0da76aab6ba23cda6cead7df5ebcdf9c74765fb)), I am seeing a test failure with a mismatch of the `ST0` value in `icicle-test/tests/x64.ins`:

```
x64.ins: 183 tests, 6 skipped, 1 errors
        [./tests/x64.ins:639] error on: line 651: expected: ST0 = 0xbfd2bec333018868 (got 0xbfd2bec333018866)
```

**Note**: Maybe this regression appeared because the test executable exits with `0` even if there is an error.

I get the same result on my M2 Macbook Pro and Linux with an AMD processor. Rust version 1.72.1

Steps to reproduce, after cloning the repo, execute the following:

```sh
$ git clone https://github.com/icicle-emu/ghidra
$ export GHIDRA_SRC=$(pwd)/ghidra
$ cd icicle-test
$ cargo run --bin icicle-test
[...]
x64.ins: 183 tests, 6 skipped, 1 errors
        [./tests/x64.ins:639] error on: line 651: expected: ST0 = 0xbfd2bec333018868 (got 0xbfd2bec333018866)
[...]
```

**Comment by mchesser** [comment]
This is flaky test related to inaccurate 80-bit float emulation. 

Currently 80-bit floats are converted to 64-bit floats for emulation. The helper function that emulates that particular function ends up calling the `exp2` library function. It appears that `exp2` rounds slightly differently with msvc (which is what I think I used for calculating the ground-truth) compared to glibc.

**Comment by rbran** [comment]
> It appears that exp2 rounds slightly differently with msvc 

That's true and it reminds me of an old bug in Team-Fortress 2 caused by gcc and msvc generating different instructions for the same source code: https://youtube.com/clip/UgkxgNNb2ay7H2eBKHkhcG54ojzHDq3JpPiL?si=okMLMHLor-Qi1L56
